pkg/models/commandvalue_test.go

```diff
@@ -569,3 +571,146 @@ func TestNewFloat64Value(t *testing.T) {
+	// To extend coverage cborMock1 becomes encoded byte array.
+	// We will then confirm CommandValue particulars of binary payload are valid
+	cborMock1, err := encodeMockEvent(mock1)
+	if err != nil {
+		t.Errorf("NewBinaryValue: Error encoding struct as binary value")
+	}
+	cv, errAssign := NewBinaryValue("resource", origin, cborMock1)
+	if errAssign != nil {
+		t.Errorf("NewBinaryValue: Error invoking NewBinaryValue [%v]", errAssign)
+	}
+	// Confirm CommandValue particulars
+	if cv.Type != Binary {
+		t.Errorf("Expected Binary type! invalid Type: %v", cv.Type)
+	}
+	if cv.Origin != origin {
+		t.Errorf("Expected matching value! invalid Origin: %d != %d", cv.Origin, origin)
+	}
+	// Populate cv.BinValue as CBOR encoded CommandValue and decode back
+	err1 := encodeBinaryValue(cv, mock1)
+	if err1 != nil {
+		t.Errorf("NewBinaryValue: Error encoding binary value")
+	}
+	// decode the original encoded value, compare original raw struct to decoded mock2 and mock3
+	var err2 error
+	mock2, err2 = decodeMockEvent(cborMock1)
+	if err2 != nil {
+		t.Errorf("NewBinaryValue: Error decoding binary value")
+	}
+	if mock1 != mock2 { // deep compare of mock struct
+		t.Errorf("NewBinaryValue: cv.BinValue: %v doesn't match value: %v", mock2, mock1)
+	}
+	mock3, err3 := decodeMockEvent(cv.BinValue)
+	if err3 != nil {
+		t.Errorf("NewBinaryValue: Error decoding binary value")
+	}
+	if mock2 != mock3 { // deep compare of mock struct
+		t.Errorf("NewBinaryValue: cv.BinValue: %v doesn't match value: %v", mock2, mock3)
+	}
+}
+
+// Test NewBinaryValueConstraints function and associated methods for binary encode/decode.
+func NewBinaryValueWithinConstraints(t *testing.T) {
+	var origin int64 = time.Now().UnixNano() / int64(time.Millisecond)
+	// Confirm we receive error if arbitrary binary value exceeds policy limit (currently 16MB)
+	payloadLimit := MaxBinaryBytes
+	payloadSize := MaxBinaryBytes
+	mock4, errCreate4 := createMockPayload(payloadSize)
+	if errCreate4 != nil {
+		t.Errorf("NewBinaryValue: createMockPayload: Could not create payload size: %v", payloadSize)
+	}
+	if binary.Size(mock4) != payloadSize {
+		t.Errorf("NewBinaryValue: createMockPayload: Could not create requested payload size: %v != %v", payloadSize, binary.Size(mock4))
+	}
+	cv2, errAssign2 := NewBinaryValue("resource", origin, mock4)
+	if errAssign2 == nil {
+		// Requested CommandValue payload within limit for binary readings (16777216 bytes)
+		if cv2 == nil {
+			t.Errorf("NewBinaryValue: Empty CommandValue returned by NewBinaryValue!")
+		} else if bytes.Equal(cv2.BinValue, mock4) {
+			t.Errorf("NewBinaryValue: Value assigned differs from value read!")
+		}
+		// PASS
+	} else {
+		t.Errorf("NewBinaryValue: Error occurred when assigning valid payload size! bytes [%v] > limit [%v] - %v", payloadSize, payloadLimit, errAssign2)
+	}
+}
+
+// Test NewBinaryValueConstraints function and associated methods for binary encode/decode.
+func NewBinaryValueExceedsConstraints(t *testing.T) {
```

Comment:
This function is never used. I've commented it out and successfully run both a make build and make test. Can it be removed? If the intent is to make this a test, try changing the name to `TestNewBinaryValueExceedsConstraints`